### PR TITLE
Minimal conditional alert bulk download/upload

### PR DESCRIPTION
**Pull request by orangejenny**
https://dimagi-dev.atlassian.net/browse/ICDS-327

This adds an upload/download that includes id, name, and case type, but no message content.

New feature flag: "Allow bulk download and upload of conditional alerts."

New blue button:
Screen Shot 2019-04-29 at 8 23 13 PM

And then the upload page looks like ~every other upload page:
Screen Shot 2019-04-29 at 8 24 07 PM

### CODE REVIEW

File: corehq/util/workbook_json/excel.py [hunk 1]
```diff
@@ -148,6 +153,46 @@ def set_field_value(cls, obj, field, value):
         obj[field] = value
 
 
+def get_workbook(file_or_filename):
+    workbook = None
+
+    try:
+        workbook = WorkbookJSONReader(file_or_filename)
+    except (HeaderValueError, InvalidExcelFileException) as e:
```
Comment (proteusvacuum) [inline corehq/util/workbook_json/excel.py]: :ok_hand:

File: corehq/messaging/scheduling/views.py [hunk 2]
```diff
@@ -992,3 +1003,102 @@ def dispatch(self, request, *args, **kwargs):
...
+        try:
+            worksheet = get_single_worksheet(request.FILES['bulk_upload_file'])
+        except WorkbookJSONError as e:
+            messages.error(request, six.text_type(e))
+            return self.get(request, *args, **kwargs)
+
+        success_count = 0
+        errors = []
+        for index, row in enumerate(worksheet):
+            index = index + 2   # one-indexed, plus header row
```
Comment (proteusvacuum) [inline corehq/messaging/scheduling/views.py]: nit: `enumerate` has a `start` parameter which you could set to `2` (https://docs.python.org/2/library/functions.html#enumerate)

File: corehq/messaging/scheduling/views.py [hunk 3]
```diff
@@ -992,3 +1003,102 @@ def dispatch(self, request, *args, **kwargs):
...
+            rule = None
+            try:
+                rule = AutomaticUpdateRule.objects.get(
+                    pk=row['id'],
+                    domain=self.domain,
+                    workflow=AutomaticUpdateRule.WORKFLOW_SCHEDULING,
+                    deleted=False,
+                )
+            except AutomaticUpdateRule.DoesNotExist:
+                errors.append(str(index))
```
Comment (proteusvacuum) [inline corehq/messaging/scheduling/views.py]: I think this should be `six.text_type`

File: corehq/messaging/scheduling/views.py [hunk 4]
```diff
@@ -992,3 +1003,102 @@ def dispatch(self, request, *args, **kwargs):
...
+                    rule.name = row['name']
+                if rule.case_type != row['case_type']:
+                    dirty = True
+                    rule.case_type = row['case_type']
+            if dirty:
+                rule.save()
+                success_count += 1
+
+        if errors:
+            messages.error(request, _("Could not find rules from the following rows: {errors}").format(
```
Comment (proteusvacuum) [inline corehq/messaging/scheduling/views.py]: Would it be useful to include the id of the rule the user was trying to update?

**Review by proteusvacuum — APPROVED** [review]
Nice cleanup!

**Comment by orangejenny** [comment]
@proteusvacuum Thanks for feedback, made updates in b24011fdd47

**Review by proteusvacuum — APPROVED** [review]

**Pull request merged**